src/equations.ml

```diff
@@ -54,8 +54,10 @@ let define_unfolding_eq ~pm env evd flags p unfp prog prog' ei hook =
                                         ++ Names.Id.print id)
         in
         let grc = UnivGen.fresh_global_instance (Global.env()) (GlobRef.ConstRef gr) in
-        Autorewrite.add_rew_rules (info.base_id ^ "_where") [CAst.make (grc, true, None)];
-        Autorewrite.add_rew_rules (info.base_id ^ "_where_rev") [CAst.make (grc, false, None)]
+        Autorewrite.add_rew_rules ~locality:Goptions.OptDefault
+          (info.base_id ^ "_where") [CAst.make (grc, true, None)];
+        Autorewrite.add_rew_rules ~locality:Goptions.OptDefault
+          (info.base_id ^ "_where_rev") [CAst.make (grc, false, None)]
```

Comment:
This is not great, it will warn in sections in a confusing way for users. Should we use the discharge providing object instead?

Reply:
ping @mattam82

Reply:
Yes !

Reply:
updated